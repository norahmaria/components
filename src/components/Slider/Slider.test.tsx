import React from 'react'
import { render } from '@testing-library/react'

import { Slider } from '../index'
import setValue from '../../utils/tests/setValue'

describe('Slider Component', () => {
  it('Can render', () => {
    const { getByTestId } = render(
      <Slider label="Slider" id="Slider" onSliderChange={() => {}} />
    )

    expect(getByTestId('slider-wrapper')).toMatchSnapshot()
  })

  it('Can run onSliderChange event on changes', () => {
    const onSliderChange = jest.fn()

    const { getByTestId } = render(
      <Slider label="Slider" id="Slider" onSliderChange={onSliderChange} />
    )

    const input = getByTestId('slider-input')
    setValue(input, '30')

    expect(input).toHaveValue('30')
    expect(onSliderChange).toHaveBeenCalledTimes(1)
  })

  it('Does not run onSliderChange events if disabled', () => {
    const onSliderChange = jest.fn()

    const { getByTestId } = render(
      <Slider
        label="Slider"
        id="Slider"
        disabled
        onSliderChange={onSliderChange}
      />
    )

    const input = getByTestId('slider-input')
    setValue(input, '30')

    expect(onSliderChange).not.toHaveBeenCalled()
  })

  it('Can color the bar by percentage', () => {
    const { getByTestId } = render(
      <Slider label="Slider" id="Slider" onSliderChange={() => {}} />
    )

    const placebo = getByTestId('slider-placebo')
    const input = getByTestId('slider-input')

    setValue(input, '30')

    expect(placebo).toHaveStyle(
      'background-image: linear-gradient(90deg, hsl(250, 100%, 67%) 30%, #E7E9EB 30%)'
    )
  })
})
