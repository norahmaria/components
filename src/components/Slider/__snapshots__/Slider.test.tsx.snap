// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Component Can render 1`] = `
<div
  class="
        slider-nm 
        size-medium
        disabled-false
        vertical-false
        undefined
      "
  data-testid="slider-wrapper"
>
  <label
    class="form-label-nm disabled-false"
    for="Slider"
  >
    Slider
  </label>
  <input
    data-testid="slider-input"
    id="Slider"
    max="100"
    min="0"
    step="1"
    type="range"
    value="20"
  />
  <div
    class="slider-nm__range color-primary"
    data-testid="slider-placebo"
  >
    <span
      class="slider-nm__dot"
      style="margin-left: 20%;"
    />
  </div>
</div>
`;
